### PR TITLE
Add QR code scanning feature

### PR DESCRIPTION
**Pull request by kevwargo**

### CODE REVIEW

**Review by tdi — APPROVED** [review]
👍

**Review by mescam — APPROVED** [review]

**Review by okonos — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -22,7 +22,19 @@ At the first execution you will be asked to set up a new password for the client
...
+
+##### Secret source
+
+if the `--secret` flag (or its short form `-s`) is omitted the value for the new secret is read from the terminal standard input without echoing the characters. If the value for the flag _is_ provided it defines the source for the new secret to be imported from. The supported forms of the flag's value are described below:
+
+- `qr-scan`: a QR code is scanned from the screen and its decoded value is used as the new secret
+- `qr-file:<IMAGE_FILE>`: a QR code is read from the `<IMAGE_FILE>` and its decoded value is used as the new secret
+- `env:<ENV>`: the secret is set to the value of the `<ENV>` environment variable
+- `file:<FILENAME>`: the secret is set to the whole contents of the file `<FILENAME>` (including a possible newline!)
+- `pass:<PLAN_TEXT>`: the secret is set to `<PLAN_TEXT>`
```
Comment (wojtii) [inline README.md]:
```suggestion
- `pass:<PLAIN_TEXT>`: the secret is set to `<PLAIN_TEXT>`
```

File: README.md [hunk 2]
```diff
@@ -22,7 +22,19 @@ At the first execution you will be asked to set up a new password for the client
...
+
+Note: the QR code scanning from the screen assumes the `import` command from the [Imagemagick](http://imagemagick.sourceforge.net/http/www/import.html) toolit is installed on the system.
```
Comment (wojtii) [inline README.md]:
```suggestion
Note: the QR code scanning from the screen assumes the `import` command from the [Imagemagick](http://imagemagick.sourceforge.net/http/www/import.html) toolkit is installed on the system.
```

File: README.md [hunk 3]
```diff
@@ -22,7 +22,19 @@ At the first execution you will be asked to set up a new password for the client
...
+After the password is set up (or if the vault already exists) you will be asked to provide a client secret which will be used to generate TOTP codes. 
+
+##### Secret source
+
+if the `--secret` flag (or its short form `-s`) is omitted the value for the new secret is read from the terminal standard input without echoing the characters. If the value for the flag _is_ provided it defines the source for the new secret to be imported from. The supported forms of the flag's value are described below:
+
+- `qr-scan`: a QR code is scanned from the screen and its decoded value is used as the new secret
+- `qr-file:<IMAGE_FILE>`: a QR code is read from the `<IMAGE_FILE>` and its decoded value is used as the new secret
+- `env:<ENV>`: the secret is set to the value of the `<ENV>` environment variable
+- `file:<FILENAME>`: the secret is set to the whole contents of the file `<FILENAME>` (including a possible newline!)
```
Comment (wojtii) [inline README.md]:
```suggestion
- `file:<FILENAME>`: the secret is set to the whole content of the file `<FILENAME>` (including a possible newline!)
```

**Review by wojtii — APPROVED** [review]
`qr-file` works without problem on macos, but `qr-scan` does not because of `import` in imagemagick resulting in:
> import: delegate library support not built-in '' (X11) @ error/import.c/ImportImageCommand/1282.

From what I have seen the option to install imagemagick with x11 support (`brew install imagemagick --with-x11`) was removed few years ago from homebrew so there is no quick fix for that 😔.

**Comment by kevwargo** [comment]
> `qr-file` works without problem on macos, but `qr-scan` does not because of `import` in imagemagick resulting in:
> 
> > import: delegate library support not built-in '' (X11) @ error/import.c/ImportImageCommand/1282.
> 
> From what I have seen the option to install imagemagick with x11 support (`brew install imagemagick --with-x11`) was removed few years ago from homebrew so there is no quick fix for that .

I think there should be some other tool for Mac to make screenshots from command line. To be researched.

**Comment by tdi** [comment]
@kevwargo screencapture is the tool

**Pull request merged**